src/screens/Ticketing/Purchase/TariffZones/index.tsx

```diff
@@ -138,128 +169,311 @@ const destinationPickerValue = (
   }
 };
 
-const TariffZones: React.FC<Props> = ({navigation, route}) => {
-  const styles = useStyles();
-  const {theme} = useTheme();
+const TariffZones: React.FC<Props> = ({navigation, route: {params}}) => {
+  const {fromTariffZone, toTariffZone} = params;
+  const [regionEvent, setRegionEvent] = useState<RegionEvent>();
+  const {tariff_zones: tariffZones} = useRemoteConfig();
+  const [selectedZones, setSelectedZones] = useState<TariffZoneSelection>({
+    from: fromTariffZone,
+    to: toTariffZone,
+    selectNext: 'to',
+  });
 
-  const {t, language} = useTranslation();
+  useEffect(() => {
+    setSelectedZones({
+      ...selectedZones,
+      from: fromTariffZone,
+      to: toTariffZone,
+    });
+  }, [fromTariffZone, toTariffZone]);
```

Comment:
Is this effect necessary? Wouldn't the zones be set in the default parameters going into the `selectedZones` `useState`?

Reply:
I think it is necessary. When selecting a zone from the venue search (TariffZoneSearch screen) it navigates back to the TariffZones screen which is already in the stack, therefore not initializing the `selectedZones` from scratch.

Reply:
Alright! In that case it definitely is necessary 👍🏻